### PR TITLE
Update frequenz dependencies

### PR DESCRIPTION
**Pull request by Marenz**

### CODE REVIEW

File: src/frequenz/dispatch/_dispatcher.py [hunk 1]
```diff
@@ -13,22 +13,22 @@
 from frequenz.dispatch._event import DispatchEvent
 from frequenz.dispatch.actor import DispatchingActor
 
-ReceivedT = TypeVar("ReceivedT")
```
Comment (Marenz) [inline src/frequenz/dispatch/_dispatcher.py:16]: The channels dependency changed to covariant types, so we have to follow suit

File: tests/test_frequenz_dispatch.py [hunk 2]
```diff
@@ -207,7 +198,7 @@ async def test_existing_dispatch_deleted(
         case Created(dispatch) | Updated(dispatch):
             assert False, "Expected a deleted event"
         case Deleted(dispatch):
-            sample.create_time = dispatch.create_time
```
Comment (Marenz) [inline tests/test_frequenz_dispatch.py:210]: Create time is now set through line 191 `sample = ...` which returns the created dispatch.

File: tests/test_frequenz_dispatch.py [hunk 3]
```diff
@@ -63,32 +62,16 @@ class ActorTestEnv:
...
     microgrid_id = randint(1, 100)
 
     actor = DispatchingActor(
         microgrid_id=microgrid_id,
         grpc_channel=MagicMock(),
         svc_addr="localhost",
-        updated_dispatch_sender=YieldingSender(updated_dispatches),
-        ready_dispatch_sender=YieldingSender(ready_dispatches),
+        updated_dispatch_sender=updated_dispatches.new_sender(),
+        ready_dispatch_sender=ready_dispatches.new_sender(),
```
Comment (shsms) [inline tests/test_frequenz_dispatch.py:74]: you don't need the sleep anymore?

**Review by shsms — COMMENTED** [review]
Just one comment/question.  Other than that, are the use of `replace` instead of setting values, and the removal of the `YieldingSender` related to the sdk update?

Either way, it might be nicer to put them in separate commits, so you can explain the reasons for them a bit more.

**Comment by Marenz** [comment]
> the use of replace instead of setting values

This is because the client-dispatch update changed `Dispatch` classes to `frozen`

> and the removal of the YieldingSender

That is a good question. As the tests show, it doesn't seem to be needed anymore, I am not sure about the exact reason though. Maybe it was never actually required to begin with and I added it to solve a problem but actually a different changed fixed it?

**Comment by shsms** [comment]
split into 3 commits, why don't you want more stats in: https://github.com/frequenz-floss/frequenz-dispatch-python/graphs/contributors

**Comment by Marenz** [comment]
> and the removal of the YieldingSender

It was removed because `send()` used to be async but isn't anymore in the latest version

File: tests/test_frequenz_dispatch.py (hunk 3 above)
Comment (Marenz) [reply tests/test_frequenz_dispatch.py:74]: I sleep very little nowadays :laughing:

**Comment by shsms** [comment]
`send()` in channels repo is still async, not sure if you mean something else, bit LGTM otherwise.

**Comment by Marenz** [comment]
> send() in channels repo is still async, not sure if you mean something else, bit LGTM otherwise.

I don't even know at this point, but it works now :shrug:

File: pyproject.toml
Comment (llucax) [inline pyproject.toml:1]: The commit "Update pyproject dependencies" has some conflict comments in the commit message.

File: src/frequenz/dispatch/_dispatcher.py (hunk 1 above)
Comment (llucax) [reply src/frequenz/dispatch/_dispatcher.py:16]: Yeah, by now maybe we should move `ReceiverFetcher` to `channels`, it proven to be useful.

File: pyproject.toml [hunk 4]
```diff
@@ -90,6 +88,7 @@ dev-pytest = [
   "pytest-asyncio == 0.23.6",
   "async-solipsism == 0.6",
   "time-machine == 2.14.1",
+  "sybil == 6.0.3",
```
Comment (llucax) [inline pyproject.toml:92]: Maybe add a comment about why `sybil` is pinned now?

**Review by llucax — COMMENTED** [review]
"Adjust code to new channels version" commit message needs an update, because as Sahas pointed out, `send()` didn't change, so the reason why `YieldingSender` is not needed anymore should be another one. If we don't know exactly I guess it is better to just say that instead.

**Review by llucax — APPROVED** [review]

**Pull request merged**